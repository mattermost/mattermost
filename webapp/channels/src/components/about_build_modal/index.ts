// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {connect} from 'react-redux';

import {getConfig, getLicense} from 'mattermost-redux/selectors/entities/general';

import {getSocketStatus} from 'selectors/views/websocket';

import type {GlobalState} from 'types/store';

import AboutBuildModal from './about_build_modal';

function mapStateToProps(state: GlobalState) {
    return {
        config: getConfig(state),
        license: getLicense(state),
        socketStatus: getSocketStatus(state),
    };
}

export default connect(mapStateToProps)(AboutBuildModal);
