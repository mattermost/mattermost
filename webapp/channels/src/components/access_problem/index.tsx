// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React, {useCallback, useEffect} from 'react';
import {useIntl} from 'react-intl';
import {useHistory} from 'react-router-dom';

import AccessProblemSVG from 'components/common/svg_images_components/access_problem_svg';
import type {CustomizeHeaderType} from 'components/header_footer_route/header_footer_route';

import './access_problem.scss';

type AccessProblemProps = {
    onCustomizeHeader?: CustomizeHeaderType;
}

const AccessProblem = ({
    onCustomizeHeader,
}: AccessProblemProps) => {
    const {formatMessage} = useIntl();
    const history = useHistory();

    const handleHeaderBackButtonOnClick = useCallback(() => {
        history.goBack();
    }, [history]);

    useEffect(() => {
        if (onCustomizeHeader) {
            onCustomizeHeader({
                onBackButtonClick: handleHeaderBackButtonOnClick,
            });
        }
    }, [onCustomizeHeader, handleHeaderBackButtonOnClick]);

    return (
        <div className='AccessProblem__body'>
            <AccessProblemSVG/>
            <div className='AccessProblem__title'>
                {formatMessage({id: 'login.contact_admin.title'})}
            </div>
            <div className='AccessProblem__description'>
                {formatMessage({id: 'login.contact_admin.detail'})}
            </div>
        </div>
    );
};

export default AccessProblem;
