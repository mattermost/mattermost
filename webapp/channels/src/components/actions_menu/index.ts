// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {ComponentProps} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import type {Dispatch} from 'redux';

import type {AppBinding} from '@mattermost/types/apps';
import type {Post} from '@mattermost/types/posts';

import {Permissions} from 'mattermost-redux/constants';
import {AppBindingLocations} from 'mattermost-redux/constants/apps';
import {appsEnabled} from 'mattermost-redux/selectors/entities/apps';
import {isMarketplaceEnabled} from 'mattermost-redux/selectors/entities/general';
import {haveICurrentTeamPermission} from 'mattermost-redux/selectors/entities/roles';
import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
import {isCombinedUserActivityPost} from 'mattermost-redux/utils/post_list';
import {isSystemMessage} from 'mattermost-redux/utils/post_utils';
import {isSystemAdmin} from 'mattermost-redux/utils/user_utils';

import {makeFetchBindings, postEphemeralCallResponseForPost, handleBindingClick, openAppsModal} from 'actions/apps';
import {openModal} from 'actions/views/modals';
import {getIsMobileView} from 'selectors/views/browser';

import type {GlobalState} from 'types/store';

import ActionsMenu from './actions_menu';
import {makeGetPostOptionBinding} from './selectors';

type Props = {
    post: Post;
    handleCardClick?: (post: Post) => void;
    handleDropdownOpened: (open: boolean) => void;
    isMenuOpen: boolean;
    location?: ComponentProps<typeof ActionsMenu>['location'];
};

const emptyBindings: AppBinding[] = [];

const getPostOptionBinding = makeGetPostOptionBinding();

const fetchBindings = makeFetchBindings(AppBindingLocations.POST_MENU_ITEM);

function mapStateToProps(state: GlobalState, ownProps: Props) {
    const {post} = ownProps;

    const systemMessage = isSystemMessage(post);

    const apps = appsEnabled(state);
    const showBindings = apps && !systemMessage && !isCombinedUserActivityPost(post.id);
    let appBindings: AppBinding[] | null = emptyBindings;
    if (showBindings) {
        appBindings = getPostOptionBinding(state, ownProps.location);
    }
    const currentUser = getCurrentUser(state);
    const isSysAdmin = isSystemAdmin(currentUser.roles);

    return {
        appBindings,
        appsEnabled: apps,
        pluginMenuItemComponents: state.plugins.components.PostDropdownMenuItem,
        isSysAdmin,
        pluginMenuItems: state.plugins.components.PostDropdownMenu,
        teamId: getCurrentTeamId(state),
        isMobileView: getIsMobileView(state),
        canOpenMarketplace: (
            isMarketplaceEnabled(state) &&
            haveICurrentTeamPermission(state, Permissions.SYSCONSOLE_WRITE_PLUGINS)
        ),
    };
}

function mapDispatchToProps(dispatch: Dispatch) {
    return {
        actions: bindActionCreators({
            handleBindingClick,
            fetchBindings,
            openModal,
            openAppsModal,
            postEphemeralCallResponseForPost,
        }, dispatch),
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(ActionsMenu);
