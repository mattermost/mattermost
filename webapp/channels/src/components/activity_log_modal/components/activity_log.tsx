// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import type {MessageDescriptor} from 'react-intl';
import {FormattedDate, FormattedMessage, FormattedTime, defineMessages} from 'react-intl';

import type {Session} from '@mattermost/types/sessions';

import {General} from 'mattermost-redux/constants';

import {getMonthLong} from 'utils/i18n';

import DeviceIcon from './device_icon';
import MoreInfo from './more_info';

type Props = {

    /**
     * The index of this instance within the list
     */
    index: number;

    /**
     * The current locale of the user
     */
    locale: string;

    /**
     * The session that's to be displayed
     */
    currentSession: Session;

    /**
     * Function to revoke session
     */
    submitRevoke: (sessionId: string, event: React.MouseEvent) => void;
};

type State = {
    moreInfo: boolean;
};

type MobileSessionInfo = {
    devicePicture?: string;
    deviceTitle?: MessageDescriptor;
    devicePlatform?: JSX.Element;
};

export default class ActivityLog extends React.PureComponent<Props, State> {
    constructor(props: Props) {
        super(props);

        this.state = {
            moreInfo: false,
        };
    }

    handleMoreInfo = (): void => {
        this.setState({moreInfo: true});
    };

    submitRevoke = (e: React.MouseEvent): void => {
        this.props.submitRevoke(this.props.currentSession.id, e);
    };

    isMobileSession = (session: Session): boolean => {
        return Boolean(session.device_id && (session.device_id.includes('apple') || session.device_id.includes('android')));
    };

    mobileSessionInfo = (session: Session): MobileSessionInfo => {
        let devicePlatform;
        let devicePicture;
        let deviceTitle;

        if (session.device_id.includes('apple')) {
            devicePicture = 'fa fa-apple';
            deviceTitle = messages.appleIcon;
            devicePlatform = (
                <FormattedMessage
                    id='activity_log_modal.iphoneNativeClassicApp'
                    defaultMessage='iPhone Native Classic App'
                />
            );

            if (session.device_id.includes(General.PUSH_NOTIFY_APPLE_REACT_NATIVE)) {
                devicePlatform = (
                    <FormattedMessage
                        id='activity_log_modal.iphoneNativeApp'
                        defaultMessage='iPhone Native App'
                    />
                );
            }
        } else if (session.device_id.includes('android')) {
            devicePicture = 'fa fa-android';
            deviceTitle = messages.androidIcon;
            devicePlatform = (
                <FormattedMessage
                    id='activity_log_modal.androidNativeClassicApp'
                    defaultMessage='Android Native Classic App'
                />
            );

            if (session.device_id.includes(General.PUSH_NOTIFY_ANDROID_REACT_NATIVE)) {
                devicePlatform = (
                    <FormattedMessage
                        id='activity_log_modal.androidNativeApp'
                        defaultMessage='Android Native App'
                    />
                );
            }
        }

        return {
            devicePicture,
            deviceTitle,
            devicePlatform,
        };
    };

    render(): React.ReactNode {
        const {
            index,
            locale,
            currentSession,
        } = this.props;

        const lastAccessTime = new Date(currentSession.last_activity_at);
        let devicePlatform = currentSession.props.platform;
        let devicePicture: string | undefined = '';
        let deviceTitle: MessageDescriptor | string = '';

        if (this.isMobileSession(currentSession)) {
            const sessionInfo = this.mobileSessionInfo(currentSession);
            devicePicture = sessionInfo.devicePicture;
            devicePlatform = sessionInfo.devicePlatform;
            deviceTitle = sessionInfo.deviceTitle || deviceTitle;
        } else {
            if (currentSession.props.platform === 'Windows') {
                devicePicture = 'fa fa-windows';
                deviceTitle = messages.windowsIcon;
            } else if (currentSession.props.platform === 'Macintosh' ||
                currentSession.props.platform === 'iPhone') {
                devicePicture = 'fa fa-apple';
                deviceTitle = messages.appleIcon;
            } else if (currentSession.props.platform === 'Linux') {
                if (currentSession.props.os.indexOf('Android') >= 0) {
                    devicePlatform = (
                        <FormattedMessage
                            id='activity_log_modal.android'
                            defaultMessage='Android'
                        />
                    );
                    devicePicture = 'fa fa-android';
                    deviceTitle = messages.androidIcon;
                } else {
                    devicePicture = 'fa fa-linux';
                    deviceTitle = messages.linuxIcon;
                }
            } else if (currentSession.props.os.indexOf('Linux') !== -1) {
                devicePicture = 'fa fa-linux';
                deviceTitle = messages.linuxIcon;
            }

            if (currentSession.props.browser.indexOf('Desktop App') !== -1) {
                devicePlatform = (
                    <FormattedMessage
                        id='activity_log_modal.desktop'
                        defaultMessage='Native Desktop App'
                    />
                );
            }
        }

        return (
            <div
                key={'activityLogEntryKey' + index}
                className='activity-log__table'
            >
                <div className='activity-log__report'>
                    <div className='report__platform'>
                        <DeviceIcon
                            devicePicture={devicePicture}
                            deviceTitle={deviceTitle}
                        />
                        {devicePlatform}
                    </div>
                    <div className='report__info'>
                        <div>
                            <FormattedMessage
                                id='activity_log.lastActivity'
                                defaultMessage='Last activity: {date}, {time}'
                                values={{
                                    date: (
                                        <FormattedDate
                                            value={lastAccessTime}
                                            day='2-digit'
                                            month={getMonthLong(locale)}
                                            year='numeric'
                                        />
                                    ),
                                    time: (
                                        <FormattedTime
                                            value={lastAccessTime}
                                            hour='2-digit'
                                            minute='2-digit'
                                        />
                                    ),
                                }}
                            />
                        </div>
                        <MoreInfo
                            locale={locale}
                            currentSession={currentSession}
                            moreInfo={this.state.moreInfo}
                            handleMoreInfo={this.handleMoreInfo}
                        />
                    </div>
                </div>
                <div className='activity-log__action'>
                    <button
                        onClick={this.submitRevoke}
                        className='btn btn-primary'
                    >
                        <FormattedMessage
                            id='activity_log.logout'
                            defaultMessage='Log Out'
                        />
                    </button>
                </div>
            </div>
        );
    }
}

const messages = defineMessages({
    androidIcon: {
        id: 'device_icons.android',
        defaultMessage: 'Android Icon',
    },
    appleIcon: {
        id: 'device_icons.apple',
        defaultMessage: 'Apple Icon',
    },
    linuxIcon: {
        id: 'device_icons.linux',
        defaultMessage: 'Linux Icon',
    },
    windowsIcon: {
        id: 'device_icons.windows',
        defaultMessage: 'Windows Icon',
    },
});
