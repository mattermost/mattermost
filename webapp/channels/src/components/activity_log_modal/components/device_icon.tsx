// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import type {MessageDescriptor} from 'react-intl';
import {useIntl} from 'react-intl';

import {formatAsString} from 'utils/i18n';

type Props = {
    devicePicture?: string;
    deviceTitle: MessageDescriptor | string;
}

export default function DeviceIcon(props: Props) {
    const intl = useIntl();

    return (
        <i
            className={props.devicePicture}
            title={formatAsString(intl.formatMessage, props.deviceTitle)}
        />
    );
}
