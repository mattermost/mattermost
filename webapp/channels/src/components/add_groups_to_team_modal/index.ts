// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import type {Dispatch} from 'redux';

import type {Group} from '@mattermost/types/groups';
import type {Team} from '@mattermost/types/teams';

import {getGroupsNotAssociatedToTeam, linkGroupSyncable, getAllGroupsAssociatedToTeam} from 'mattermost-redux/actions/groups';
import {getGroupsNotAssociatedToTeam as selectGroupsNotAssociatedToTeam} from 'mattermost-redux/selectors/entities/groups';
import {getCurrentTeam, getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';

import {setModalSearchTerm} from 'actions/views/search';

import type {GlobalState} from 'types/store';

import AddGroupsToTeamModal from './add_groups_to_team_modal';

type Props = {
    team?: Team;
    skipCommit?: boolean;
    onAddCallback?: (groupIDs: string[]) => void;
    excludeGroups?: Group[];
}

function mapStateToProps(state: GlobalState, ownProps: Props) {
    const searchTerm = state.views.search.modalSearch;

    const team = ownProps.team || getCurrentTeam(state);

    let groups = selectGroupsNotAssociatedToTeam(state, team?.id || '');
    if (searchTerm) {
        const regex = RegExp(searchTerm, 'i');
        groups = groups.filter((group) => regex.test(group.display_name) || regex.test(group.name));
    }

    return {
        currentTeamName: team?.display_name,
        currentTeamId: team?.id ?? getCurrentTeamId(state),
        skipCommit: ownProps.skipCommit,
        onAddCallback: ownProps.onAddCallback,
        excludeGroups: ownProps.excludeGroups,
        searchTerm,
        groups,
    };
}

function mapDispatchToProps(dispatch: Dispatch) {
    return {
        actions: bindActionCreators({
            getGroupsNotAssociatedToTeam,
            setModalSearchTerm,
            linkGroupSyncable,
            getAllGroupsAssociatedToTeam,
        }, dispatch),
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(AddGroupsToTeamModal);
