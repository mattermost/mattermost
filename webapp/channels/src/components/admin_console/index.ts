// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {connect} from 'react-redux';
import type {ConnectedProps} from 'react-redux';
import {bindActionCreators} from 'redux';
import type {Dispatch} from 'redux';

import {getConfig, getEnvironmentConfig, patchConfig} from 'mattermost-redux/actions/admin';
import {loadRolesIfNeeded, editRole} from 'mattermost-redux/actions/roles';
import {selectTeam} from 'mattermost-redux/actions/teams';
import {General} from 'mattermost-redux/constants';
import * as Selectors from 'mattermost-redux/selectors/entities/admin';
import {getConfig as getGeneralConfig, getLicense} from 'mattermost-redux/selectors/entities/general';
import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
import {getRoles} from 'mattermost-redux/selectors/entities/roles';
import {getTeam} from 'mattermost-redux/selectors/entities/teams';
import {isCurrentUserSystemAdmin, currentUserHasAnAdminRole, getCurrentUserId} from 'mattermost-redux/selectors/entities/users';

import {setNavigationBlocked, deferNavigation, cancelNavigation, confirmNavigation} from 'actions/admin_actions.jsx';
import {setAdminConsoleUsersManagementTableProperties} from 'actions/views/admin';
import {selectLhsItem} from 'actions/views/lhs';
import {getAdminDefinition, getConsoleAccess} from 'selectors/admin_console';
import {showNavigationPrompt} from 'selectors/views/admin';
import LocalStorageStore from 'stores/local_storage_store';

import type {GlobalState} from 'types/store';

import AdminConsole from './admin_console';

function mapStateToProps(state: GlobalState) {
    const generalConfig = getGeneralConfig(state);
    const buildEnterpriseReady = generalConfig.BuildEnterpriseReady === 'true';
    const adminDefinition = getAdminDefinition(state);
    const teamId = LocalStorageStore.getPreviousTeamId(getCurrentUserId(state));
    const team = getTeam(state, teamId || '');
    const unauthorizedRoute = team ? `/${team.name}/channels/${General.DEFAULT_CHANNEL}` : '/';
    const consoleAccess = getConsoleAccess(state);

    return {
        config: Selectors.getConfig(state),
        environmentConfig: Selectors.getEnvironmentConfig(state),
        license: getLicense(state),
        buildEnterpriseReady,
        unauthorizedRoute,
        showNavigationPrompt: showNavigationPrompt(state),
        isCurrentUserSystemAdmin: isCurrentUserSystemAdmin(state),
        currentUserHasAnAdminRole: currentUserHasAnAdminRole(state),
        roles: getRoles(state),
        adminDefinition,
        consoleAccess,
        cloud: state.entities.cloud,
        team,
        currentTheme: getTheme(state),
    };
}

function mapDispatchToProps(dispatch: Dispatch) {
    return {
        actions: bindActionCreators({
            getConfig,
            getEnvironmentConfig,
            patchConfig,
            setNavigationBlocked,
            deferNavigation,
            cancelNavigation,
            confirmNavigation,
            loadRolesIfNeeded,
            editRole,
            selectLhsItem,
            selectTeam,
            setAdminConsoleUsersManagementTableProperties,
        }, dispatch),
    };
}

const connector = connect(mapStateToProps, mapDispatchToProps);

export type PropsFromRedux = ConnectedProps<typeof connector>;

export default connector(AdminConsole);
