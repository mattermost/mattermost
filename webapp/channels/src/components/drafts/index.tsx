// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React, {memo, useEffect, useMemo} from 'react';
import {useDispatch, useSelector} from 'react-redux';

import {getCurrentUser} from 'mattermost-redux/selectors/entities/common';
import {getTeammateNameDisplaySetting} from 'mattermost-redux/selectors/entities/preferences';
import {isScheduledPostsEnabled} from 'mattermost-redux/selectors/entities/scheduled_posts';
import {getStatusForUserId} from 'mattermost-redux/selectors/entities/users';
import {displayUsername} from 'mattermost-redux/utils/user_utils';

import {selectLhsItem} from 'actions/views/lhs';
import {suppressRHS, unsuppressRHS} from 'actions/views/rhs';
import type {Draft} from 'selectors/drafts';
import {makeGetDrafts} from 'selectors/drafts';

import DraftList from 'components/drafts/draft_list';

import type {GlobalState} from 'types/store';
import {LhsItemType, LhsPage} from 'types/store/lhs';

import DraftsAndSchedulePostsPageHeader from './drafts_and_schedule_posts_page_header';
import DraftsAndSchedulePostsTabs from './drafts_and_schedule_posts_tabs';

import './drafts_and_schedule_posts.scss';

const EMPTY_DRAFTS: Draft[] = [];

function Drafts() {
    const dispatch = useDispatch();

    const scheduledPostsEnabled = useSelector(isScheduledPostsEnabled);

    // We would need to get drafts here early since its needed by the draft list component
    const getDrafts = useMemo(() => makeGetDrafts(), []);
    const drafts = useSelector(getDrafts);

    const currentUser = useSelector(getCurrentUser);
    const userStatus = useSelector((state: GlobalState) => getStatusForUserId(state, currentUser.id));

    const teammateNameDisplaySetting = useSelector(getTeammateNameDisplaySetting);
    const userDisplayName = useMemo(() => displayUsername(currentUser, teammateNameDisplaySetting), [currentUser, teammateNameDisplaySetting]);

    // When Drafts component mounts, select Drafts in the LHS
    // and suppress the RHS and restore RHS when component unmounts
    useEffect(() => {
        dispatch(selectLhsItem(LhsItemType.Page, LhsPage.Drafts));
        dispatch(suppressRHS);

        return () => {
            dispatch(unsuppressRHS);
        };
    }, [dispatch]);

    if (scheduledPostsEnabled) {
        return (
            <DraftsAndSchedulePostsPageHeader>
                <DraftsAndSchedulePostsTabs
                    drafts={drafts || EMPTY_DRAFTS}
                    currentUser={currentUser}
                    userDisplayName={userDisplayName}
                    userStatus={userStatus}
                />
            </DraftsAndSchedulePostsPageHeader>
        );
    }

    return (
        <DraftsAndSchedulePostsPageHeader>
            <DraftList
                drafts={drafts || EMPTY_DRAFTS}
                currentUser={currentUser}
                userDisplayName={userDisplayName}
                userStatus={userStatus}
            />
        </DraftsAndSchedulePostsPageHeader>
    );
}

export default memo(Drafts);
