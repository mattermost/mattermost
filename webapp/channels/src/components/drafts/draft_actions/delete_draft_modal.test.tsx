// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';
import {Provider} from 'react-redux';

import {GenericModal} from '@mattermost/components';

import mockStore from 'tests/test_store';

import DeleteDraftModal from './delete_draft_modal';

describe('components/drafts/draft_actions/delete_draft_modal', () => {
    const baseProps = {
        displayName: 'display_name',
        onConfirm: jest.fn(),
        onExited: jest.fn(),
    };

    it('should match snapshot', () => {
        const store = mockStore();

        const wrapper = shallow(
            <Provider store={store}>
                <DeleteDraftModal
                    {...baseProps}
                />
            </Provider>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should have called onConfirm', () => {
        const wrapper = shallow(
            <DeleteDraftModal {...baseProps}/>,
        );

        wrapper.find(GenericModal).first().props().handleConfirm!();
        expect(baseProps.onConfirm).toHaveBeenCalledTimes(1);
        expect(wrapper).toMatchSnapshot();
    });

    it('should have called onExited', () => {
        const wrapper = shallow(
            <DeleteDraftModal {...baseProps}/>,
        );

        wrapper.find(GenericModal).first().props().onExited?.();
        expect(baseProps.onExited).toHaveBeenCalledTimes(1);
        expect(wrapper).toMatchSnapshot();
    });
});
