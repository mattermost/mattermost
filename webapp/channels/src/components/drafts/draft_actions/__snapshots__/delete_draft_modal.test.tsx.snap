// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/drafts/draft_actions/delete_draft_modal should have called onConfirm 1`] = `
<GenericModal
  autoCloseOnCancelButton={true}
  autoCloseOnConfirmButton={true}
  autoFocusConfirmButton={true}
  bodyPadding={true}
  compassDesign={true}
  confirmButtonText="Yes, delete"
  enforceFocus={true}
  handleCancel={[Function]}
  handleConfirm={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  id="genericModal"
  isDeleteModal={true}
  keyboardEscape={true}
  modalHeaderText="Delete draft"
  onExited={[MockFunction]}
  show={true}
>
  <MemoizedFormattedMessage
    defaultMessage="Are you sure you want to delete this draft to <strong>{displayName}</strong>?"
    id="drafts.confirm.delete.text"
    values={
      Object {
        "displayName": "display_name",
        "strong": [Function],
      }
    }
  />
</GenericModal>
`;

exports[`components/drafts/draft_actions/delete_draft_modal should have called onExited 1`] = `
<GenericModal
  autoCloseOnCancelButton={true}
  autoCloseOnConfirmButton={true}
  autoFocusConfirmButton={true}
  bodyPadding={true}
  compassDesign={true}
  confirmButtonText="Yes, delete"
  enforceFocus={true}
  handleCancel={[Function]}
  handleConfirm={[MockFunction]}
  id="genericModal"
  isDeleteModal={true}
  keyboardEscape={true}
  modalHeaderText="Delete draft"
  onExited={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  show={true}
>
  <MemoizedFormattedMessage
    defaultMessage="Are you sure you want to delete this draft to <strong>{displayName}</strong>?"
    id="drafts.confirm.delete.text"
    values={
      Object {
        "displayName": "display_name",
        "strong": [Function],
      }
    }
  />
</GenericModal>
`;

exports[`components/drafts/draft_actions/delete_draft_modal should match snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <DeleteDraftModal
    displayName="display_name"
    onConfirm={[MockFunction]}
    onExited={[MockFunction]}
  />
</ContextProvider>
`;
